index.js

```diff
@@ -1,3 +1,10 @@
 'use strict'; // eslint-disable-line strict
 
+/**
+ * Catch uncaught exceptions and add timestamp to aid debugging
+ */
+process.on('uncaughtException', err => {
+    process.stderr.write(`${new Date().toISOString()}: Uncaught exception: \n${err.stack}`);
+});
```

Comment:
This will change the service behavior in case of error right? Like, we won't crash, and we won't end the current request either, is that what we want? Maybe I'm missing something here?

Reply:
That's what I thought too, but from observation he worker still exits after the message is printed.

Reply:
old PR, but the worker exits because he adds its own listener (https://github.com/scality/cloudserver/blob/4f7aa548866068608fc534b3eca6f266583c599c/lib/server.js#L82-L91) to shutdown after this one.

However the primary  (that uses fork and has the monitoring server for prom metrics) don't have another listener, so this is keep the primary up in undefined state if an exception is caught